### ISSUE TITLE
Using `networknt / json-schema-validator` for the backend validation of `jsonSchema { }`?

### ISSUE THREAD

**Issue by takanuva15** [kotest/kotest#3734]
Hi, I was just wondering whether the kotest team had already considered using the [`networknt/json-schema-validator`](https://github.com/networknt/json-schema-validator/blob/master/doc/quickstart.md) library as the backend for its jsonSchema validation support? `json-schema-validator` supports the popular jackson JsonNode class natively, and it already supports important features like `$ref`, recursive schemas, and array conditions which get used widely in enterprise architectures.

I'm no expert on kotest's internal workings, but I imagine it would be pretty simple to have Kotest's `jsonSchema { }` dsl pipe its output to the json-schema-validator's Factory, so then kotest wouldn't need to redefine all the internal logic for handling `$ref` and array conditions that `json-schema-validator` already has implemented and running in prod.

**Comment by Kantis** [comment]
I agree that there's a lot of gritty details of json-schemas that we simply don't have support for currently, and which could be resolved as you suggest. The major drawback of going that route is that the suggested library is JVM-only, so we would drop support for other platforms. Perhaps it could be worth it, or perhaps we could build an `expect / actual` layer around it to use our simplified implemenetation for non-JVM targets. 

Personally I'm leaning towards keeping it as-is until there's significant demand for unimplemented features.
